abi/realloc.go

```diff
@@ -0,0 +1,52 @@
+package abi
+
+import "unsafe"
+
+// realloc allocates or reallocates memory for Component Model calls across
+// the host-guest boundary.
+//
+// Note: the use of uintptr assumes 32-bit pointers, e.g. GOOS=wasm32 when compiled for WebAssembly.
+//
+//go:export cabi_realloc
+//go:wasmexport cabi_realloc
+func realloc(ptr unsafe.Pointer, size, align, newsize uintptr) unsafe.Pointer {
```

Comment:
What happens if allocation failed due to out of memory issue?

Reply:
Both Go and TinyGo panic with out of memory if they can't allocate a block.

Reply:
Is a runtime panic acceptable to the Component Model if it can't allocate memory?

Reply:
@ydnar There is no way to capture that allocation failure and retry or notify the caller.  It's an immediate, uncatchable, runtime panic().  We don't have a choice in this case.

Reply:
A runtime failure seems reasonable to me.